lib/routes/account.js

```diff
@@ -864,6 +869,63 @@ module.exports = function (
       }
     },
     {
+      method: 'POST',
+      path: '/account/devices/notify',
+      config: {
+        auth: {
+          strategy: 'sessionTokenWithDevice'
+        },
+        validate: {
+          payload: isA.alternatives().try(
+            isA.object({
+              to: isA.string().valid('all').required(),
+              excluded: isA.array().items(isA.string().length(32).regex(HEX_STRING)).optional(),
+              payload: isA.object().required(),
+              TTL: isA.number().integer().min(0).optional()
+            }),
+            isA.object({
+              to: isA.array().items(isA.string().length(32).regex(HEX_STRING)).required(),
+              payload: isA.object().required(),
+              TTL: isA.number().integer().min(0).optional()
+            })
+          )
+        },
+        response: {
+          schema: {}
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.devicesNotify', request)
+        var body = request.payload
+        var sessionToken = request.auth.credentials
+        var uid = sessionToken.uid
+        var payload = body.payload
+
+        var schemaPath = path.resolve(__dirname, PUSH_PAYLOADS_SCHEMA_PATH)
+        readFile(schemaPath).then(function(schema) {
+          schema = JSON.parse(schema)
```

Comment:
This will read the file on every request, it's probably worth reading it once on startup and storing the results in a global somewhere.

:heart: validation here though!